master/internal/task/allocation.go

```diff
@@ -660,7 +672,12 @@ func (a *Allocation) RestoreResourceFailure(
 	if a.req.Restore {
 		switch heartbeat := cluster.TheLastBootClusterHeartbeat(); {
 		case a.model.StartTime == nil:
-			break
+			// This happens in k8s when submitted but not running pods get deleted
+			// when master is away. We never record a start time for this allocation
+			// and we will want to close it. This is an issue since we keep trying
+			// to restore these commands without an endTime.
+			a.model.StartTime = ptrs.Ptr(time.Now().UTC())
+			a.model.EndTime = a.model.StartTime
```

Comment:
this is slightly confusing to me, this would look like the allocation ran for no time at all where previously they would look like they never started at all. is there a way to fix this on the other end?

Reply:
Added a filter to both trial / command restore checking that the allocation state was not terminated instead